pkg/directory/gh.go

```diff
@@ -0,0 +1,363 @@
+func NewGitHub(org string) (*GH, error) {
+	ctx := context.Background()
+	client := &GH{}
+
+	token, ok := os.LookupEnv("GITHUB_TOKEN")
+	if !ok {
+		return client, errors.New("GITHUB_TOKEN not set")
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client.Client = github.NewClient(tc)
+	client.UsersService = client.Client.Users
+	client.Org = org
+
+	if err := client.getMembersAndTeams(); err != nil {
+		return client, err
+	}
+	return client, nil
+}
+
+func (g *GH) getMembersAndTeams() error {
+	update := false
+
+	_, err := os.Stat(cacheDir)
+	if err != nil {
+		os.MkdirAll(cacheDir, os.ModePerm)
+	}
+
+	membersFile := path.Join(cacheDir, "members")
+	mfInfo, err := os.Stat(membersFile)
+	if err != nil || time.Since(mfInfo.ModTime()).Minutes() > cacheTTL {
+		update = true
+	}
+
+	teamsFile := path.Join(cacheDir, "teams")
+	tfInfo, err := os.Stat(teamsFile)
+	if err != nil || time.Since(tfInfo.ModTime()).Minutes() > cacheTTL {
+		update = true
+	}
+
+	if update {
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			g.getMembers()
+		}()
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			g.getTeams()
+		}()
+		wg.Wait()
+
+		if err := saveCache(membersFile, g.Members); err != nil {
+			return err
+		}
+		if err := saveCache(teamsFile, g.Teams); err != nil {
+			return nil
+		}
+	} else {
+		if err := getCached(membersFile, &g.Members); err != nil {
+			return err
+		}
+		if err := getCached(teamsFile, &g.Teams); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func saveCache(filename string, v interface{}) error {
+	buf, err := json.Marshal(v)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("unable to marshal cache file %s", filename))
+	}
+
+	if _, err := os.Stat(filename); os.IsExist(err) {
+		if err := os.Remove(filename); err != nil {
+			return err
+		}
+	}
+
+	if err := ioutil.WriteFile(filename, buf, 0700); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("unable to write cache file %s", filename))
+	}
+	return nil
+}
+
+func getCached(filename string, v interface{}) error {
+	_, err := os.Stat(filename)
+	if err != nil {
+		return err
+	}
+
+	buf, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("unable to read cached file %s", filename))
+	}
+
+	if err := json.Unmarshal(buf, v); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("unable to unmarshal cache file: %s", filename))
+	}
+
+	return nil
+}
+
+func (g *GH) getMembers() error {
+	members := []Member{}
+
+	in := make(chan string)
+	out := make(chan Member)
+
+	// This process can be slow so we speed it up by doing multiple lookups at a time.
+	// Was implemented because it took about 45 seconds to get all members and teams and this
+	// took it down to about 3 seconds.
+	wg := sync.WaitGroup{}
+	for i := 0; i < workers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for {
+				login, ok := <-in
```

Comment:
The idiomatic way to read from a channel until it is closed is to use a for/range loop: `for login := range in`